fix: dev build fails on file change

<!--Thanks for taking the time to fill out this issue! Please fill out the sections below to the best of your ability. If you don't know the answer to a section, that's okay, just leave it blank. -->
<!--Before you submit, please search open issues to see if your bug has already been reported. -->

## Description
The dev server break on file changes.

```
CLI Watching for changes in "."
CLI Ignoring changes in "**/{.git,node_modules}/**" | "dist"
Cleaning up react source ...
CLI Change detected: change src/components/form/form.ts
ESM Build start
✘ [ERROR] Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/collection/index.ts"

✘ [ERROR] Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/code/index.ts"

✘ [ERROR] Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/details/index.ts"

✘ [ERROR] Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/form/index.ts"

✘ [ERROR] Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/index.ts"

ESM Build failed
Error: Build failed with 5 errors:
error: Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/collection/index.ts"
error: Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/code/index.ts"
error: Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/details/index.ts"
error: Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/form/index.ts"
error: Could not resolve "/home/jaeaeich/Desktop/cloud-components/packages/ecc-utils-design/src/react/index.ts"
    at failureErrorWithLog (/home/jaeaeich/Desktop/cloud-components/node_modules/tsup/node_modules/esbuild/lib/main.js:1651:15)
    at /home/jaeaeich/Desktop/cloud-components/node_modules/tsup/node_modules/esbuild/lib/main.js:1059:25
    at runOnEndCallbacks (/home/jaeaeich/Desktop/cloud-components/node_modules/tsup/node_modules/esbuild/lib/main.js:1486:45)
    at buildResponseToResult (/home/jaeaeich/Desktop/cloud-components/node_modules/tsup/node_modules/esbuild/lib/main.js:1057:7)
    at /home/jaeaeich/Desktop/cloud-components/node_modules/tsup/node_modules/esbuild/lib/main.js:1086:16
    at responseCallbacks.<computed> (/home/jaeaeich/Desktop/cloud-components/node_modules/tsup/node_modules/esbuild/lib/main.js:704:9)
    at handleIncomingPacket (/home/jaeaeich/Desktop/cloud-components/node_modules/tsup/node_modules/esbuild/lib/main.js:764:9)
    at Socket.readFromStdout (/home/jaeaeich/Desktop/cloud-components/node_modules/tsup/node_modules/esbuild/lib/main.js:680:7)
    at Socket.emit (node:events:518:28)
    at addChunk (node:internal/streams/readable:559:12)
```

--- a/scripts/build.js
+++ b/scripts/build.js
@@ -151,7 +151,12 @@ nextTask("Building source", async () => {
     .finally(() =>
       nextTask("Cleaning up react source", () => {
         if (!commanderOpts.watch) {
-          nextTask("Cleaning up react source", () => {
+if (!commanderOpts.watch) {
+  fs.rmSync(reactDir, {
+    force: true,
+    recursive: true,
+  });
+}
             fs.rmSync(reactDir, {
               force: true,
               recursive: true,

--- a/scripts/build.js
+++ b/scripts/build.js
@@ -148,18 +148,17 @@ nextTask("Building source", async () => {
         );
       });
     })
-    .finally(() =>
-      nextTask("Cleaning up react source", () => {
-        if (!commanderOpts.watch) {
-          nextTask("Cleaning up react source", () => {
-            fs.rmSync(reactDir, {
-              force: true,
-              recursive: true,
-            });
+    .finally(() => {
+      // Skip cleanup if watch mode is enabled to preserve the react source for incremental builds
+      if (!commanderOpts.watch) {
+        nextTask("Cleaning up react source", () => {
+          fs.rmSync(reactDir, {
+            force: true,
+            recursive: true,
           });
-        }
-      })
-    );
+        });
+      }
+    });
 });
 
 module.exports = {};

--- a/scripts/build.js
+++ b/scripts/build.js
@@ -150,10 +150,14 @@ nextTask("Building source", async () => {
     })
     .finally(() =>
       nextTask("Cleaning up react source", () => {
-        fs.rmSync(reactDir, {
-          force: true,
-          recursive: true,
-        });
+        if (!commanderOpts.watch) {
+          nextTask("Cleaning up react source", () => {
+            fs.rmSync(reactDir, {
+              force: true,
+              recursive: true,
+            });
+          });
+        }
       })
     );
 });

--- a/scripts/build.js
+++ b/scripts/build.js
@@ -151,12 +151,7 @@ nextTask("Building source", async () => {
     .finally(() =>
       nextTask("Cleaning up react source", () => {
         if (!commanderOpts.watch) {
-if (!commanderOpts.watch) {
-  fs.rmSync(reactDir, {
-    force: true,
-    recursive: true,
-  });
-}
+          nextTask("Cleaning up react source", () => {
             fs.rmSync(reactDir, {
               force: true,
               recursive: true,

